shotover-proxy/src/transforms/redis/sink_cluster.rs

```diff
@@ -108,6 +115,25 @@ impl RedisSinkCluster {
         Ok(sink_cluster)
     }
 
+    async fn direct_connection(&mut self) -> Result<&UnboundedSender<Request>> {
```

Comment:
Im a bit conflicted on if caching the direct connection in self.direct_connection is the right call here.

reasons to cache:
Its more correct as we can maintain connection state for the direct connection for the lifetime of the chain.
Bonus: faster lookup.

reasons not to cache:
maybe we shouldnt be relying on connection state in the first place as that state wont exist when querying a SLOT off of this connection.
bonus: the connection pool will auto-detect and reopen closed connections for us.

At the moment I'm kind of thinking it doesnt matter a whole lot either way right now and we should properly address this when we look into connection state issues like RESP3 support.

Reply:
I'm ok to revisit this properly when we look into resp3